SM64O/Form1.cs

```diff
@@ -966,21 +966,68 @@ private void button2_Click(object sender, EventArgs e)
 
         private void Form1_Load(object sender, EventArgs e)
         {
-            comboBox1.SelectedIndex = 0;
-            comboBox2.SelectedIndex = 0;
+            comboBoxEmulator.SelectedIndex = 0;
+            comboBoxChar.SelectedIndex = 0;
             gamemodeBox.SelectedIndex = 0;
 
             Settings sets = Settings.Load("settings.xml");
 
             if (sets != null)
             {
-                textBox5.Text = sets.LastIp;
+                textBoxAddress.Text = sets.LastIp;
                 numericUpDown2.Value = sets.LastPort;
                 usernameBox.Text = sets.Username;
 
-                comboBox1.SelectedIndex = sets.LastEmulator;
-                comboBox2.SelectedIndex = sets.LastCharacter;
+                comboBoxEmulator.SelectedIndex = sets.LastEmulator;
+                comboBoxChar.SelectedIndex = sets.LastCharacter;
             }
+           
+            // Create the ToolTip and associate with the Form container.
+            ToolTip toolTip1 = new ToolTip();
+
+            // Set up the delays for the ToolTip.
+            toolTip1.AutoPopDelay = 5000;
+            toolTip1.InitialDelay = 1000;
+            toolTip1.ReshowDelay = 500;
+            // Force the ToolTip text to be displayed whether or not the form is active.
+            toolTip1.ShowAlways = true;
+
+            // Set up the ToolTip text for the Buttons, Labels and Checkboxes.
+            // Could probably use better names for the labels, buttons and checkboxes, to tell them apart!
+            toolTip1.SetToolTip(this.labelAddress, "Input the IP Address to the host");
+            toolTip1.SetToolTip(this.textBoxAddress, "Input the IP Address to the host");
+
+            toolTip1.SetToolTip(this.labelPort, "Input the port to the host");
+            toolTip1.SetToolTip(this.numericUpDown2, "Input the port to the host");
+
+            toolTip1.SetToolTip(this.labelUsername, "Input your username");
+            toolTip1.SetToolTip(this.usernameBox, "Input your username");
+
+            toolTip1.SetToolTip(this.checkBoxChat, "Check this to disable the chat in your server");
+            toolTip1.SetToolTip(this.checkBoxServer, "Check this if you want to make your own server");
+            toolTip1.SetToolTip(this.checkBoxLAN, "Check this to only allow LAN connections to your server");
```

Comment:
Please change this to add "Check this to disable UPnP and port checking service" or similar